### PR TITLE
[App Service] BREAKING CHANGE: Update websites SDK to the latest version (azure-mgmt-web==2.0.0) & Adopt track2 SDK

### LINKED ISSUE #17124: az cli "hybrid connections not supported on linux app."
**Describe the bug**
Since hybrid-connections are now supported on Linux webapps via the portal, it should also be supported via CLI,
The following warning message appears: "hybrid connections not supported on linux app." and it does not add the hybrid-connection.

**To Reproduce**
execute command on linux webapp: 
az webapp hybrid-connection add --hybrid-connection
                                --name
                                --namespace
                                --resource-group
                                [--slot]
                                [--subscription]

**Expected behavior**
A hybrid connection should be added to the linux webapp

**Environment summary**
```
Linux-4.15.0-1106-azure-x86_64-with-debian-10.2 (Cloud Shell)
Python 3.6.10
Installer: DEB

azure-cli 2.19.0 *
```

**Additional context**
Documentation does not mention that this does not work on linux webapps

### LINKED ISSUE #17228: WebApp:az webapp deploy only works when extension is installed
**Describe the bug**

In CLI v2.20.0, the `az webapp deploy` command only works if the user also has the `webapp` extension installed.

Command without the extension installed:
```
jason@Azure:~$ az webapp deploy --resource-group freebergDemo --name jboss-eap-bug-bash --type war --src-url https://tomcat.apache.org/tomcat-9.0-doc/appdev/sample/sample.war
This command is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
The command failed with an unexpected error. Here is the traceback:
Object of type 'frozenset' is not JSON serializable
Traceback (most recent call last):
    File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 239, in invoke
    self.output.out(cmd_result, formatter=formatter, out_file=out_file)
    File "/opt/az/lib/python3.6/site-packages/knack/output.py", line 148, in out
    output = formatter(obj)
    File "/opt/az/lib/python3.6/site-packages/knack/output.py", line 40, in format_json
    separators=(',', ': ')) + '\n'
    File "/opt/az/lib/python3.6/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
    File "/opt/az/lib/python3.6/json/encoder.py", line 201, in encode
    chunks = list(chunks)
    File "/opt/az/lib/python3.6/json/encoder.py", line 430, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
    File "/opt/az/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
    File "/opt/az/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
    File "/opt/az/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
    File "/opt/az/lib/python3.6/json/encoder.py", line 437, in _iterencode
    o = _default(o)
    File "/opt/az/lib/python3.6/site-packages/knack/output.py", line 32, in default
    return json.JSONEncoder.default(self, o)
    File "/opt/az/lib/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'frozenset' is not JSON serializable
```

When the `webapps` extension is installed, the command works correctly. There is some sort of interaction between the base CLI in v2.20.0 and the webapps extension. When the extension **is** installed, the logs say that the behavior of the command is augmented by the presence of the webapps extension.

**To Reproduce**

Have v2.20.0 installed and try deploying a file from a URL: `az webapp deploy --resource-group Linux --name linux-java8-jboss-7 --type war --src-url https://tomcat.apache.org/tomcat-9.0-doc/appdev/sample/sample.war`

**Expected behavior**

The user shouldn't need to have the webapps extension installed for the `deploy` command to work correctly.

**Environment summary**

- az version 2.20.0
- Tested using Azure Cloud Shell

**Additional context**

### PR DESCRIPTION
**Pull request by panchagnula**
Fixes #17228
Fixes #17124
**Description**
Updating websites extension to latest SDK https://docs.microsoft.com/en-us/python/api/azure-mgmt-web/azure.mgmt.web.v2020_09_01?view=azure-python & this also has the track 2 SDK changes as well.
https://pypi.org/project/azure-mgmt-web/2.0.0/

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/requirements.py3.Darwin.txt
Comment (yungezz) [inline src/azure-cli/requirements.py3.Darwin.txt:83]: just curious, why downgrading this version? It might need live test of synapse tests also.

File: src/azure-cli/requirements.py3.Darwin.txt
Comment (yungezz) [inline src/azure-cli/requirements.py3.Darwin.txt:82]: [thumb up]!

File: src/azure-cli/requirements.py3.Darwin.txt
Comment (panchagnula) [reply src/azure-cli/requirements.py3.Darwin.txt:83]: This was a change from rebasing to the upstream/dev before starting PR

**Comment by yungezz** [comment]
since this is a major SDK version change from Track1 to Track2, I suspect there should be code change also. And run live tests for impacted command modules to avoid regression. thanks.

**Comment by panchagnula** [comment]
> HI @panchagnula , since this is a major SDK version change from Track1 to Track2, there should be code change also. And run live tests for impacted command modules to avoid regression. thanks.
> 
> * Track2 migration guide https://github.com/Azure/azure-cli/blob/dev/doc/track_2_migration_guidance.md
>   cc @qwordy

yes the PR is marked as draft while i get the tests re-recorded make the necessary fixes.

**Comment by qwordy** [comment]
@panchagnula Could you please summarize the code change pattern when updating to new SDK? Thanks a lot

**Comment by panchagnula** [comment]
> @panchagnula Could you please summarize the code change pattern when updating to new SDK? Thanks a lot

@qwordy Sorry, I am not sure what 'code change pattern' are you referring to? Please elaborate or we can discuss offline in email if that helps. Thank you.

**Comment by qwordy** [comment]
> 
> 
> > @panchagnula Could you please summarize the code change pattern when updating to new SDK? Thanks a lot
> 
> @qwordy Sorry, I am not sure what 'code change pattern' are you referring to? Please elaborate or we can discuss offline in email if that helps. Thank you.

@panchagnula This article lists typical modifications when updating to new SDK. Appreciate if you can summarize modifications you made in this PR.
https://github.com/Azure/azure-cli/blob/dev/doc/track_2_migration_guidance.md

**Comment by qwordy** [comment]
@panchagnula This PR isn't active for about one month and is in draft status. Do you encounter any problem? We can provide help.

**Comment by AnatoliB** [comment]
Tagging @gzuber for Functions

**Comment by panchagnula** [comment]
@qwordy can you help me find an owner from bot service - seems like they have their own version of webapp_operations they use & this will be impacted by the new change - so they need fix these on their end.

**Comment by panchagnula** [comment]
@qwordy can you review this change & provide feedback as needed. Thank you.

File: src/azure-cli/azure/cli/command_modules/appservice/_create_util.py
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/_create_util.py:340]: If it is debugging purpose, use logger. If it is command output, consider returning data in a method. Azure CLI will print them. We seldom print data by self.

File: src/azure-cli/azure/cli/command_modules/appservice/_validators.py
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/_validators.py:28]: logger is better

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_validators.py
Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/_validators.py:28]: Thanks for pointing this out - I had this while I was trying to debug. Will remove this one.

File: scripts/sdk_process/patch_models.py
Comment (fengzhou-msft) [inline scripts/sdk_process/patch_models.py:236]:
```suggestion
        'azure.mgmt.loganalytics',
        'azure.mgmt.web'
```

Comment (panchagnula) [reply scripts/sdk_process/patch_models.py:236]: @fengzhou-msft Ah! I will .web - log analytics isn't an extension I own; I hope its still ok for me to update this as well?

**Review by fengzhou-msft — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]
Approved for databoxedge part

**Review by msyyc — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_appservice_utils.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_appservice_utils.py:33]: please avoid using print() in azure cli and try to use logger https://github.com/microsoft/knack/blob/dev/docs/logging.md

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:176]: is there an alternative parameter to use? If yes, please ad redirect info in your deprecation object.

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py
Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:176]: No there is no alternate property

**Review by Juliehzl — APPROVED** [review]

**Comment by panchagnula** [comment]
@qwordy can we merge this once the build with the conflicting change merge is successful? Thanks!

**Pull request merged**
